moveit_ros/planning/kinematics_plugin_loader/src/kinematics_plugin_loader.cpp

```diff
@@ -287,78 +295,101 @@ robot_model::SolverAllocatorFn KinematicsPluginLoader::getLoaderFunction(const s
+            base_param_name = robot_description_ + "_kinematics." + known_groups[i].name_;
+            RCLCPP_INFO(LOGGER_KINEMATICS_PLUGIN_LOADER, "Looking for param %s ", (base_param_name + ".kinematics_solver").c_str());
+
+            auto parameters_and_prefixes = ksolver_params->get_parameters({ base_param_name + ".kinematics_solver" });
+            for (auto & value : parameters_and_prefixes) {
+                ksolver_param_name = value.value_to_string();
+                found = true;
+                std::stringstream ss(ksolver_param_name);
+                bool first = true;
+                while (ss.good() && !ss.eof())
                 {
-                  first = false;
-                  groups_.push_back(known_groups[i].name_);
+                  if (first)
+                  {
+                    first = false;
+                    groups_.push_back(known_groups[i].name_);
+                  }
+                  std::string solver;
+                  ss >> solver >> std::ws;
+                  possible_kinematics_solvers[known_groups[i].name_].push_back(solver);
+                  RCLCPP_INFO(LOGGER_KINEMATICS_PLUGIN_LOADER, "Using kinematics solver '%s' for group '%s'.", solver.c_str(),
+                                  known_groups[i].name_.c_str());
                 }
-                std::string solver;
-                ss >> solver >> std::ws;
-                possible_kinematics_solvers[known_groups[i].name_].push_back(solver);
-                ROS_DEBUG_NAMED(LOGNAME, "Using kinematics solver '%s' for group '%s'.", solver.c_str(),
-                                known_groups[i].name_.c_str());
-              }
             }
+            // rclcpp::Parameter param =  ksolver_params->get_parameter(base_param_name + ".kinematics_solver");
+            // ksolver_param_name = ksolver_params->get_parameter(base_param_name + ".kinematics_solver").get_value<std::string>();
+            // found = ksolver_params->get_parameter(base_param_name + ".kinematics_solver", parameter);
+            RCLCPP_INFO(LOGGER_KINEMATICS_PLUGIN_LOADER, "%d param is %s ", found, ksolver_param_name.c_str());
+
           }
+          // if (found)
+          // {
+          //   RCLCPP_INFO(LOGGER_KINEMATICS_PLUGIN_LOADER, "Found param %s", ksolver_param_name.c_str());
+          //   std::string ksolver;
+          //   if (ksolver_params->has_parameter(ksolver_param_name))
+          //   {
+          //     ksolver = node->get_parameter(ksolver_param_name).get_value<std::string>();
+          //
+          //   }
+          // }
 
           std::string ksolver_timeout_param_name;
-          if (nh.searchParam(base_param_name + "/kinematics_solver_timeout", ksolver_timeout_param_name))
+          if (ksolver_params->has_parameter(base_param_name + "/kinematics_solver_timeout"))
           {
+            ksolver_timeout_param_name = node->get_parameter(base_param_name + "/kinematics_solver_timeout").get_value<std::string>();
```

Comment:
`Parameter::get_value()` will throw a `ParameterTypeException` if the requested type is different from the actual `ParameterType` (http://docs.ros2.org/dashing/api/rclcpp/parameter__value_8hpp_source.html). Apparently, this happens for both `Node::get_parameter()` as well as the parameters client's version. I think if we really want to have a safe version of querying parameters, we have the following options:
1. Wrap any typed parameter queries inside try/catch blocks and handle unexpected results
2. Check if parameter exists, check parameter type, get parameter value using parameter type

If we have flexible parameter types (i.e. `int`, `double` like in the example below) we should be able to just use auto type deduction or let the default type conversion handle it.